### ISSUE TITLE
Ios delayed render

### ISSUE THREAD

**Issue by thomaspaniagua** [NativeScript/NativeScript#4412]
Hey guys, so I have this code to apply a linear gradient to views in my nativescript app. It works perfectly on android, but on Ios it won't apply the gradient unless I delay calling the linearGradient function by around 100ms. I usually call it in the (loaded) callback on a function and if on Ios I need to apply the delay there. I have also noticed that my view in ios does not render as soon as the app is launched, it takes a little bit more for my components to appear which could be part of the issue. Any clues as to what's going on?

I'm using Nativescript 3.0.3
```

import * as coreView from "ui/core/view";
import * as platform from "platform";
import * as colorModule from "color";
var Color = colorModule.Color;

type StartPoint = "UP" | "DOWN" | "LEFT" | "RIGHT";

/** Native Definitions **/
declare var NSMutableArray;
declare var interop;
declare var CAGradientLayer;
declare var CGPointMake;

export class Gradients {


    public static linearGradientById(root: any, id: string, start: StartPoint, hex?: string[]): void {
        hex = hex || ["#F41618", "#F61D6C"];
        this._linearGradient(coreView.getViewById(root, id), start, hex);
    }

    public static linearGradient(view: coreView.ViewBase, start: StartPoint, hex?: string[]): void {
        hex = hex || ["#F41618", "#F61D6C"];
        this._linearGradient(view, start, hex);
    }

    private static _linearGradient(_view: coreView.ViewBase, start: StartPoint, colors: any[]) {
        let Points = {
            Up: "UP",
            Down: "DOWN",
            Left: "LEFT",
            Right: "RIGHT"
        };

        var _colors = [],
            nativeView;

        if (_view) {
            nativeView = _view.nativeView;
        } else {
            throw Error("Cannot find view '" + _view + "' in page!");
        }

        if (!nativeView) {
            return;
        }

        colors.forEach(function (c, idx) {
            if (!(c instanceof Color)) {
                colors[idx] = new Color(c);
            }
        });

        if (platform.device.os === platform.platformNames.android) {
            let orientations = android.graphics.drawable.GradientDrawable.Orientation;

            var backgroundDrawable = nativeView.getBackground(),
                orientation,
                LINEAR_GRADIENT = 0;

            switch (start) {
                case Points.Up:
                    orientation = orientations.TOP_BOTTOM;
                    break;

                case Points.Down:
                    orientation = orientations.BOTTOM_TOP;
                    break;

                case Points.Left:
                    orientation = orientations.LEFT_RIGHT;
                    break;

                case Points.Right:
                    orientation = orientations.RIGHT_LEFT;
                    break;
            }

            colors.forEach(function (c) {
                _colors.push(c.android);
            });

            if (!(backgroundDrawable instanceof android.graphics.drawable.GradientDrawable)) {
                backgroundDrawable = new android.graphics.drawable.GradientDrawable();
                backgroundDrawable.setColors(_colors);
                backgroundDrawable.setGradientType(LINEAR_GRADIENT);
                backgroundDrawable.setOrientation(orientation);
                nativeView.setBackgroundDrawable(backgroundDrawable);
            }
        } else if (platform.device.os === platform.platformNames.ios) {
            var colorsArray = NSMutableArray.alloc().initWithCapacity(2);
            colors.forEach(function (c) {
                colorsArray.addObject(interop.types.id(c.ios.CGColor));
            });
            var gradientLayer = CAGradientLayer.layer();
            gradientLayer.colors = colorsArray;
            gradientLayer.frame = nativeView.bounds;

            switch (start) {
                case Points.Up:
                    gradientLayer.startPoint = CGPointMake(0.5, 1);
                    gradientLayer.endPoint = CGPointMake(0.5, 0);
                    break;

                case Points.Down:
                    gradientLayer.startPoint = CGPointMake(0.5, 0);
                    gradientLayer.endPoint = CGPointMake(0.5, 1);
                    break;

                case Points.Left:
                    gradientLayer.startPoint = CGPointMake(0, 0.5);
                    gradientLayer.endPoint = CGPointMake(1, 0.5);
                    break;

                case Points.Right:
                    gradientLayer.startPoint = CGPointMake(1, 0.5);
                    gradientLayer.endPoint = CGPointMake(0, 0.5);
                    break;
            }

            nativeView.layer.insertSublayerAtIndex(gradientLayer, 0);
        }
    }

}

```

**Comment by tsonevn** [comment]
Hi @thomaspaniagua,
I tested the given code for the gradient and found that you should apply timeout of 0 milliseconds for iOS, which will allow applying the gradient. For you convenience, I am attaching sample project and screenshot from the iOS simulator
[Archive.zip](https://github.com/NativeScript/NativeScript/files/1084981/Archive.zip)
.
![lk3lt26voumbtmarjgg0xkblmmouamnfg_yi79ycb3xhzk4et718hmf4dos0tqsdkdpwsjangipeui2ecgocyn0iq-_zmiwljnqsu6s4bo0f1kr3dmujedqc16ubuu85](https://user-images.githubusercontent.com/17448734/27286607-756ad9c2-5509-11e7-9a0b-21a666d431b3.png)

**Comment by thomaspaniagua** [comment]
@tsonevn 

Hey so I tried the setTimeout with 0 as delay and it only worked when I didn't have a lot going on in the view. I did find another solution (still a hack) that could lead to the actual problem. Turns out that when called in the loaded event in iOS the nativeView.frame.size height and width are both 0. What I did was delay the function call for as long as they were 0. So the question is why are they 0?
